internal/api/service/service_test.go

```diff
@@ -29,12 +31,28 @@ func newMockK8sClient(name string, time time.Time) *mockK8sClient {
 	}
 }
 
-func TestListReleasesPopulatesFromConfigMap(t *testing.T) {
-	name := "configmap-1"
+type mockGitClient struct {
+	Org    string
+	Checks github.ChecksService
+}
+
+func newMockGitClient(org string) *mockGitClient {
+	return &mockGitClient{
+		Org: org,
+	}
+}
+
+func (m *mockGitClient) GetTravisCIBuildURLForRef(ctx context.Context, repo string, ref string) (string, error) {
+	return "www.travisci.com", nil
```

Comment:
This isn't really a test mock, it's just a stubbed out implementation of the `TravisChecker` interface. A mock needs to have dynamic behaviour so that we can test different cases like, for example, failing to find a build URL for the repo/ref. 

A pattern to look out for in the future, is when a test case "magically" knows an important detail about the mock implementation. In this case, the `TestTravisFieldIsPopulated` test knows about the magic string `"www.travisci.com"`, it's a sign our test case is really just testing the mock's implementation details.

Reply:
I have changed the mock git client to use the testify mock library. I have also added a test case for when the travis build URL is not fetched.